packages/payments-plugin/src/paypal/paypal.shop-resolver.ts

```diff
@@ -0,0 +1,37 @@
+import { Mutation, Resolver } from '@nestjs/graphql';
+import {
+    ActiveOrderService,
+    Allow,
+    Ctx,
+    IllegalOperationError,
+    Permission,
+    RequestContext,
+    Transaction,
+} from '@vendure/core';
+
+import { PayPalOrder } from './graphql/generated-shop-types';
+import { PayPalOrderService } from './paypal-order.service';
+
+@Resolver()
+export class PayPalShopResolver {
+    constructor(
+        private readonly activeOrderService: ActiveOrderService,
+        private readonly paypalService: PayPalOrderService,
+    ) {}
+
+    @Mutation()
+    @Transaction()
+    @Allow(Permission.Owner)
+    async createPayPalOrder(@Ctx() ctx: RequestContext): Promise<PayPalOrder | undefined> {
+        const sessionOrder = await this.activeOrderService.getActiveOrder(ctx, {});
+
+        if (!sessionOrder) {
+            throw new IllegalOperationError('Session has no active order');
+        }
+        if (sessionOrder.state !== 'ArrangingPayment') {
+            throw new IllegalOperationError('Order must be in arranging payment state');
+        }
```

Comment:
I think these could be internationalized with existing error message for example there are:

```
    "NO_ACTIVE_ORDER_ERROR": "There is no active Order associated with the current session",
```

and

```
    "ORDER_MODIFICATION_ERROR": "Order contents may only be modified when in the \"AddingItems\" state",
    "ORDER_PAYMENT_STATE_ERROR": "A Payment may only be added when Order is in \"ArrangingPayment\" state",
    "ORDER_STATE_TRANSITION_ERROR": "Cannot transition Order from \"{ fromState }\" to \"{ toState }\"",
```

Reply:
Really good feeback, thank you :) Will definetly do that!